addons/pos_self_order_razorpay/models/pos_payment_method.py

```diff
@@ -20,5 +20,8 @@ def _payment_request_from_kiosk(self, order):
     def _load_pos_self_data_domain(self, data):
         domain = super()._load_pos_self_data_domain(data)
         if data['pos.config'][0]['self_ordering_mode'] == 'kiosk':
-            domain = expression.OR([[('use_payment_terminal', '=', 'razorpay')], domain])
+            domain = expression.OR([
+                [('use_payment_terminal', '=', 'razorpay'), ('id', 'in', data['pos.config'][0]['payment_method_ids'])],
+                domain
+            ])
```

Comment:
Can adapt adyen and stripe to use a domain like that inside their module folder.

For instance its managed in `odoo/addons/pos_self_order/models/pos_payment_method.py` line 14 :)